src/javascript/function-scanner.ts

```diff
@@ -31,7 +31,12 @@ export class FunctionScanner implements JavaScriptScanner {
       visit: (visitor: Visitor) => Promise<void>) {
     const visitor = new FunctionVisitor(document);
     await visit(visitor);
-    return {features: Array.from(visitor.functions)};
+    return {
+      features: Array.from(visitor.functions)
+                    .sort(
+                        (a, b) => compareSourceLocationStart(
```

Comment:
You could to `.sort(compareSourceLocationStart)` if `compareSourceLocationStart` took features not SourceLocations.